dll/win32/shell32/CDefView.cpp

```diff
@@ -1097,16 +1376,22 @@ HRESULT CDefView::FillList()
     DPA_DestroyCallback( hdpa, fill_list, this);
 
     /* sort the array */
-    if (m_pSF2Parent)
-    {
-        m_pSF2Parent->GetDefaultColumn(NULL, (ULONG*)&m_sortInfo.nHeaderID, NULL);
-    }
-    else
+    int sortCol = -1;
+    if (!IsRefreshCommand) // Are we loading for the first time?
     {
-        FIXME("no m_pSF2Parent\n");
+        m_sortInfo.bIsAscending = TRUE;
+        sortCol = 0; // In case the folder does not know/care
+        if (m_pSF2Parent)
+        {
+            ULONG foldSortCol = sortCol;
+            HRESULT hr = m_pSF2Parent->GetDefaultColumn(NULL, &foldSortCol, NULL);
+            if (SUCCEEDED(hr))
+                hr = MapFolderColumnToListColumn(foldSortCol);
```

Comment:
It would be better here (in regards to my previous comments) to use the `foldSortCol` variable here. (Maybe using a different name because it's not "old" now...)

Reply:
It is not "old" (f-old), I'll change the name to make it clear.